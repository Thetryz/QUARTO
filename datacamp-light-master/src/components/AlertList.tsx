import * as React from "react";

import { IAlert } from "../redux/view";
import Alert from "./Alert";

export interface IAlertListProps extends React.Props<AlertList> {
  alerts: IAlert[];
}

export class AlertList extends React.Component<IAlertListProps> {
  public static defaultProps: Partial<IAlertListProps> = {
    alerts: [],
  };

  public render() {
    return <div>{this.props.alerts.map(alert => <Alert {...alert} />)}</div>;
  }
}

export default AlertList;
