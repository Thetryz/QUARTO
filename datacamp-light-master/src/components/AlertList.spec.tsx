import * as React from "react";
import { shallow } from "enzyme";

import { AlertList } from "./AlertList";

describe("AlertList", () => {
  it("should exist", () => {
    expect(AlertList).toBeDefined();
  });

  it("should match snapshot", () => {
    const tree = shallow(<AlertList />);
    expect(tree).toMatchSnapshot();
  });
});
