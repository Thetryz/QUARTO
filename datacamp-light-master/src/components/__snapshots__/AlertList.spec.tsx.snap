// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertList should match snapshot 1`] = `<div />`;
